### DISCUSSION TITLE
Purpose of `z-index` in Prism?

### DISCUSSION THREAD

**Discussion by garretwilson** [PrismJS/prism#3580] [Q&A] [answered]
I understand the general purpose of `z-index` in CSS. Why specifically is it used in Prism?

I just ran into a situation where some of my drop-down menus were coming down _behind_ a Prism code block. I know how to work around that, but I'm curious why Prism felt the need to set a `z-index` to begin with, instead of just using the default `auto`?

**Comment by RunDevelopment** [comment, marked as answer]
We only have 2 themes that use `z-index`.

Coy needs `z-index: 1` because of its shadows. Here's how it looks like without the z index (or `z-index: auto`):
![image](https://user-images.githubusercontent.com/20878432/198001019-e407f462-06b9-41a8-afd5-72421a3e76b7.png)

Twilight does a little trickery with Line Highlight. It places its tokens above the highlighting box to use it as a background.

The Toolbar and Previewers plugins also use `z-index` to make their elements appear above code blocks.

**Comment by garretwilson** [comment]
> Coy needs `z-index: 1` because of its shadows.

Ah, those darn, buggy, problematic Coy shadows again. I've turned them off discussed in #2164 and #2170.

In any case I added a really big `z-index` to my CSS library to get the menus to go on top of Prism.

**Reply by RunDevelopment** [reply]
> those darn, buggy, problematic Coy shadows again

You can say that again.
